web/js/components/layer/measurement-row.js

```diff
@@ -266,27 +266,65 @@ class LayerRow extends React.Component {
     const { activeSourceIndex } = this.state;
     const sources = lodashValues(measurement.sources);
 
+    // handle activeSourceIndex values not valid after projection change
+    let validIndexSelected = false;
+    let minValidIndex = -1;
+
     const Tabs = (
       <Nav vertical className="source-tabs col-md-3 col-sm-12">
         {sources
           .sort((a, b) => a.title.localeCompare(b.title))
           .map(
-            (source, index) =>
-              hasMeasurementSetting(measurement, source)
-                ? this.renderSourceTabs(
+            (source, index) => {
+              let hasSetting = hasMeasurementSetting(measurement, source);
+
+              if (hasSetting) {
+                // check if activeSourceIndex is satisfied to set flag
+                if (activeSourceIndex === index) {
+                  validIndexSelected = true;
+                }
+                // only set minValidIndex once
+                if (minValidIndex < 0) {
+                  minValidIndex = index;
+                }
+              }
+              return [source, hasSetting];
+            }
+          )
+          .map(
```

Comment:
@edplato Looks good. Only feedback is that this could probably be all in one `map()` to prevent the double looping. Though that shouldn't make a visible difference in this case since the array shouldn't ever be very large.

Reply:
Good recommendation, I slimmed down my changes to only one `map()` and removed some extra logic.

Reply:
Nice! looks good